src/main/java/com/epam/controller/TraineeController.java

```diff
@@ -0,0 +1,133 @@
+package com.epam.controller;
+
+import com.epam.dto.trainee.*;
+import com.epam.dto.trainer.TrainerProfile;
+import com.epam.mappers.TraineeMapper;
+import com.epam.model.Trainee;
+import com.epam.model.Trainer;
+import com.epam.model.Training;
+import com.epam.service.TraineeService;
+import com.epam.service.TrainerService;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.transaction.Transactional;
+import javax.validation.Valid;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/trainee")
+@RequiredArgsConstructor
+public class TraineeController {
+    private final TraineeMapper mapper;
+    private final TraineeService traineeService;
+    private final TrainerService trainerService;
+
+    @PostMapping
+    public ResponseEntity<TraineeRegistrationResponse> registerTrainee(@RequestBody @Valid TraineeRegistrationRequest request) {
+        Trainee trainee = mapper.traineeRegistrationRequestToTrainee(request);
+        Trainee createdTrainee = traineeService.createTrainee(trainee);
```

Comment:
Yes I think you are right in spring core task there is written 'In the case that already exists Trainer or Trainee with the same pair of first and last name as a suffix to the username should be added a serial number. ' So in this case if I'll try to register it will add serial number to username but as I understand if firstname and lastname are same it shouldn't be possible as you said

Reply:
Garik jan, you probably need to do the following. If someone tries to register as a Trainee and enters user name like "Garik Martirosyan", at first you'll check if there is already registered Trainee with this username (we have a workaround for this, by adding some numbers to the end of username, right?) and then you check if there is a **Trainer** with the same username. And if there is one Trainer with the same username - you'll reject current registration and say "You cannot be Trainer and Trainee at once" or something like this. And maybe you'll need at first check for the second condition (if Trainer with the same name exists) and only then check if Trainee with the same name exists and do the workaround, otherwise it's not so effective...

Reply:
The same is for Trainer registration. Check if Trainee with the same name exists and if it's true, reject current registration. If not, proceed to check if a Trainer with the same name exists and if yes - do the workaround for them.